src/components/dock/dock-button/dock-button.tsx

```diff
@@ -37,6 +37,12 @@ export class DockButton {
     @Event()
     private itemSelected: EventEmitter<DockItem>;
 
+    /**
+     * Fired when a dock menu has been clicked from the dock.
+     */
+    @Event()
+    private dockMenuOpened: EventEmitter<DockItem>;
```

Comment:
`dockMenuOpened` or `menuOpened`?

I'm not sure why we named it `item.dockMenu` and not just `item.menu`. So `dockMenuOpened` is consistent with that, but feels inconsistent alongside `itemSelected` (why not `dockItemSelected`). 

Not sure what to suggest here, that's why I've stayed away from discussing the event name before 🙂

Reply:
I chose **_dockMenuOpened_** to make it consistent with _item.**dockMenu**._  Don't you think the event **_itemSelected_** is consistent with _item.**selected**_ ? 

But we can have another PR to refactor if we need to change the properties in the Interface

Reply:
Yes, I agree that's consistent in a way, but I don't understand why the `dock` prefix is used here and not on `itemSelected`.

Renaming the event is not refactoring, it would be a breaking change. That's why I don't want to rush this.

> Don't you think the event **_itemSelected_** is consistent with _item.**selected**_

Yes, but it doesn't help if you can't apply the same rule to name the new event that's reflected in `item.dockMenu.menuOpen`.

I can see naming the event `menuOpen` is also an option.

```jsx
<limel-dock
  dockItems={this.dockItems}
  onItemSelected={this.handleSelected} 
  onMenuOpen={this.handleMenuOpen}
/>
```

What do you think @adrianschmidt ?

Reply:
We also have the `close` event that we could have named `menuClose` or `dockMenuClosed` I guess 🤔

Reply:
Yeah, I'm with you on this being inconsistent. I feel like that was maybe discussed on the [original PR](https://github.com/Lundalogik/lime-elements/pull/1792), but there was a lot of discussion there, and I tired of searching before finding anything. I'm not sure if it was ever discussed and there was a good reason or if it was just something that was missed.

> Renaming the event is not refactoring, it would be a breaking change. That's why I don't want to rush this.

Yes, and I agree with this sentiment, but I think limel-dock is still marked as private, so the docs for it are hidden, and we are free to break it as we wish, until we "publish it" 👍 Whether you use that to justify merging this PR without deciding on the naming first, or use it to justify changing the names of existing properties and events, that's up to you 😊

Reply:
I think `menuOpen` is a good name for the event, but I'm not deeply invested either way 😅

Reply:
The event name `menuOpen` can be a good name actually. It actually go with `item.dockMenu.menuOpen`. I agree with this.